releases/release-4.0.15.md

```diff
@@ -0,0 +1,125 @@
+---
+title: TiDB 4.0.15 Release Notes
+category: Releases
+---
+
+# TiDB 4.0.15 Release Notes
+
+Release Date: September 27, 2021
+
+TiDB version: 4.0.15
+
+## Improvements
+
++ TiDB
+
+    - Trigger auto-analyze based on histogram row count [#26706](https://github.com/pingcap/tidb/pull/26706)
+
++ TiKV
+
+    - separate read write ready to reduce read latency [#10619](https://github.com/tikv/tikv/pull/10619)
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix wrong charset and collation for case when function [#26671](https://github.com/pingcap/tidb/pull/26671)
+    - Fix the issue that greatest(datetime) union null returns empty string [#26564](https://github.com/pingcap/tidb/pull/26564)
+    - Fix the issue that sometimes fails to send requests if there are tombstone stores. [#25849](https://github.com/pingcap/tidb/pull/25849)
+
```

Comment:
- Fix range building for binary literal.[26455](https://github.com/pingcap/tidb/pull/26455)
- Fix "index out of range" error when a SQL contains both group by and union.[26553](https://github.com/pingcap/tidb/pull/26553)
- executor: fix unexpected behavior when casting invalid string to date.[27935](https://github.com/pingcap/tidb/pull/27935)
- planner: add missing column for Apply convert to Join.[27282](https://github.com/pingcap/tidb/pull/27282)
- Fix bug that count disctinct on multi-columns return wrong result when new collation is on.[27830](https://github.com/pingcap/tidb/pull/27830)
- expression: fix extract bug when argument is a negative duration.[27369](https://github.com/pingcap/tidb/pull/27369)
- Make `group_concat` function consider the collation.[27835](https://github.com/pingcap/tidb/pull/27835)
- Revert [#21045](https://github.com/pingcap/tidb/pull/21045) to avoid show session variables is very slow on first run. [#24326](https://github.com/pingcap/tidb/issues/24326).[26240](https://github.com/pingcap/tidb/pull/26240)
- Fix expression rewrite makes between expr infers wrong collation.[27851](https://github.com/pingcap/tidb/pull/27851)
- Fix wrong selection push down when having above agg.[27741](https://github.com/pingcap/tidb/pull/27741)